lib/resty/etcd/v3.lua

```diff
@@ -688,43 +690,74 @@ local function request_chunk(self, method, path, opts, timeout)
 
 
     local function read_watch()
+        body = nil
 
         while(1) do
-            body, err = res.body_reader()
+            local chunk, error = res.body_reader()
+            if error then
+                return nil, error
+            end
+            if not chunk then
+                break
+            end
+
             if not body then
-                return nil, err
+                body = chunk
+            else
+                body = tab_concat({body, chunk})
             end
-            if not utils.is_empty_str(body) then
+
+            if not utils.is_empty_str(chunk) and str_byte(chunk, -1) == str_byte("\n") then
```

Comment:
So if the response chunk doesn't contain the `\n` and we'll continue reading the next chunk? But if the connection is aborted / timed out this time, so the last chunk was missed?